Diverse Source Files, Edit Translations

simstern opened the pull request:
Hi @mrimann, I did several enhancements on your package as I will use it for the base functionality of another package that overrides the Xliff files to a separate directory instead of physically overwriting it. (The package is not yet published.)
Besides the added features, this PR contains several changes also on the structure (XliffTranslationService) to connect it with my new package. Also I had to fix some logic on how the files are loaded and saved, so that it supports untranslated files with only target tags.
Let me know if you intend to merge it, or if you have some comments or improvements. I'm happy to adjust it.
I also put it to WIP for the moment, since possibly some adjustments will follow. Anyway, please have a look and let me know your thoughts.

simstern committed:
[TASK] Extract functionality into service

Functionality is extracted into a service class for re-usability in external controllers.

simstern committed:
[TASK] Flow 3.0 compatibility

Apply ViewHelper raw formatting

simstern committed:
[TASK] Service interface for overwrites in other packages

simstern committed:
[FEATURE] Diverse source files

Make not only the 'Main.xlf' but all .xlf files available for translation.

simstern committed:
[FEATURE] Edit language file

Edit a source file for a specific language. It takes the language either from the target or the source tag, depending on the source-language attribute of the xliff file.

This also fixes the following previous problems:
* source-language and target-language tag not set correctly on save translation
* xliff files with only target language (usually for default language) not correctly loaded

simstern committed:
[FEATURE] HTML text support

* CDATA for HTML strings
* Textarea for editing

simstern committed:
[TASK] Move Xliff save function to service

mrimann commented:
Hi
thanks for your contribution! I had a first look into the proposed changes and I think they point into the right direction. I'll need another few days to find time to review in detail and merge if still happy with it after detailed reading.

Just some questions out of curiousity:
- why not aim for Flow v4.x compatibility?
- could you also modify the Flow version dependency (at the moment there's no version constraint)
- what exactly will the other package you mentioned do on top of this package? And could that maybe "put together" into one package?

Cheers,
Mario

simstern committed:
[TASK] Specify required minimum Flow version

simstern commented:
Hi Mario,
Thanks for your response.

> why not aim for Flow v4.x compatibility?
The project I'm working on is currently still in v3, so my aim was simply to get it run there. However, 

I'm happy to upgrade it to v4 if this PR goes into the master. 

> could you also modify the Flow version dependency (at the moment there's no version constraint)

Done

> what exactly will the other package you mentioned do on top of this package? And could that maybe "put together" into one package?

It's public now: https://github.com/sinso/Swisscom.UserXliffTranslator, please see the readme there. I've been thinking about putting it together as well, but as it changes the default structure of how Xliff are stored, I decided to separate it. It keeps it more flexible for the different use cases.

Cheers,
Simon

mrimann commented:
Great, thanks Simon! I'll have a look into it and hopefully merge it soon.

simstern committed:
[TASK] Flush XmlModelCache on translation change

Required for production as translation files are only monitored in non-production context.
